server/admin/admin_server_test.go

```diff
@@ -188,96 +188,125 @@ func TestAdminServer_CreateTree(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 
-	invalidTree := *testonly.LogTree
+	invalidTree := *stestonly.LogTree
 	invalidTree.TreeState = trillian.TreeState_HARD_DELETED
 
+	nilPrivateKeyTree := *stestonly.LogTree
+	nilPrivateKeyTree.PrivateKey = nil
+
+	demoPrivateKey, err := keys.NewFromPrivatePEM(testonly.DemoPrivateKey, testonly.DemoPrivateKeyPass)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	tests := []struct {
 		desc string
 		req  *trillian.CreateTreeRequest
-		// shouldCreate indicates whether AdminTX.CreateTree() is expected to be called.
-		shouldCreate bool
+		// createTree is the tree that is expected to be passed to AdminTX.CreateTree().
+		// If nil, no call to AdminTX.CreateTree() is expected.
+		createTree *trillian.Tree
 		// createErr is the error that will be returned by the mock AdminTX.CreateTree() method.
 		createErr error
-		opts      testOptions
+		opts      setupOptions
 	}{
 		{
-			desc:         "validTree",
-			req:          &trillian.CreateTreeRequest{Tree: testonly.LogTree},
-			shouldCreate: true,
-			opts: testOptions{
+			desc:       "validTree",
+			req:        &trillian.CreateTreeRequest{Tree: stestonly.LogTree},
+			createTree: stestonly.LogTree,
+			opts: setupOptions{
 				shouldBeginTx: true,
 				shouldCommit:  true,
 			},
 		},
 		{
-			desc:         "invalidTree",
-			req:          &trillian.CreateTreeRequest{Tree: &invalidTree},
-			shouldCreate: true,
-			createErr:    errors.New("CreateTree failed"),
-			opts: testOptions{
+			desc:       "invalidTree",
+			req:        &trillian.CreateTreeRequest{Tree: &invalidTree},
+			createTree: &invalidTree,
+			createErr:  errors.New("CreateTree failed"),
+			opts: setupOptions{
 				shouldBeginTx: true,
 			},
 		},
 		{
-			desc:         "commitError",
-			req:          &trillian.CreateTreeRequest{Tree: testonly.LogTree},
-			shouldCreate: true,
-			opts: testOptions{
+			desc:       "commitError",
+			req:        &trillian.CreateTreeRequest{Tree: stestonly.LogTree},
+			createTree: stestonly.LogTree,
+			opts: setupOptions{
 				shouldBeginTx: true,
 				shouldCommit:  true,
 				commitErr:     errors.New("commit error"),
 			},
 		},
 		{
-			desc: "non-existent key and SignerFactory does not implement keys.Generator",
-			req:  &trillian.CreateTreeRequest{Tree: testonly.LogTree},
-			opts: testOptions{
-				signerFactory: ktestonly.NewSignerFactoryWithErr(terrors.New(terrors.NotFound, "key not found")),
+			desc: "nilPrivateKeyAndNoKeyGenerator",
+			req:  &trillian.CreateTreeRequest{Tree: &nilPrivateKeyTree},
```

Comment:
wantErr is inferred from the setupOptions (see line 352). It's probably better to make it explicit in the test case though, so I can change that.